package/cpp/api/JsiSkTextStyle.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <jsi/jsi.h>
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation"
+
+#include "Paragraph.h"
+#include "ParagraphStyle.h"
+
+#pragma clang diagnostic pop
+
+namespace RNSkia {
+
+namespace jsi = facebook::jsi;
+
+using namespace skia::textlayout; // NOLINT
+
+/**
+ Implementation of the TextStyle object in JSI for the paragraph builder
+ */
+class JsiSkTextStyle {
+public:
+  static TextStyle fromValue(jsi::Runtime &runtime, const jsi::Value &value) {
+
+    TextStyle retVal;
+
+    // Accept undefined && null
+    if (value.isUndefined() || value.isNull()) {
+      return retVal;
+    }
+
+    // Read values from the argument - expected to be a TextStyle shaped object
+    if (!value.isObject()) {
+      throw jsi::JSError(runtime, "Expected SkTextStyle as first argument");
+    }
+
+    auto object = value.asObject(runtime);
+
+    if (object.hasProperty(runtime, "backgroundColor")) {
+      auto propValue = object.getProperty(runtime, "backgroundColor");
+      SkPaint p;
+      p.setColor(JsiSkColor::fromValue(runtime, propValue));
+      retVal.setBackgroundPaint(p);
+    }
+    if (object.hasProperty(runtime, "color")) {
+      auto propValue = object.getProperty(runtime, "color");
+      retVal.setColor(JsiSkColor::fromValue(runtime, propValue));
+    }
+    if (object.hasProperty(runtime, "decoration")) {
+      auto propValue = object.getProperty(runtime, "decoration");
+      retVal.setDecoration(static_cast<TextDecoration>(propValue.asNumber()));
+    }
+    if (object.hasProperty(runtime, "decorationColor")) {
+      auto propValue = object.getProperty(runtime, "decorationColor");
+      retVal.setDecorationColor(JsiSkColor::fromValue(runtime, propValue));
+    }
+    if (object.hasProperty(runtime, "decorationThickness")) {
+      auto propValue = object.getProperty(runtime, "decorationThickness");
+      retVal.setDecorationThicknessMultiplier(propValue.asNumber());
+    }
+    if (object.hasProperty(runtime, "decorationStyle")) {
+      auto propValue = object.getProperty(runtime, "decorationStyle");
+      retVal.setDecorationStyle(
+          static_cast<TextDecorationStyle>(propValue.asNumber()));
+    }
+    if (object.hasProperty(runtime, "fontFamilies")) {
+      auto propValue = object.getProperty(runtime, "fontFamilies")
+                           .asObject(runtime)
+                           .asArray(runtime);
+      auto size = propValue.size(runtime);
+      std::vector<SkString> families(size);
+      for (size_t i = 0; i < size; ++i) {
+        families[i] = propValue.getValueAtIndex(runtime, i)
+                          .asString(runtime)
+                          .utf8(runtime)
+                          .c_str();
+      }
```

Comment:
```suggestion
      }
      retVal.setFontFamilies(families);
```

Reply:
This fixes the usage of custom fonts for us.

Reply:
Done! Thanks, @canfan